From a57846956605c87849eda5ed4e3dc0f6f761e7ee Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:18 -0400
Subject: [ethernet] qlcnic: Validate Tx queue only for 82xx adapters

Message-id: <1408626422-3941-35-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88576
O-Subject: [PATCH RHEL 7.1 034/138] qlcnic: Validate Tx queue only for 82xx adapters.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Himanshu Madhani <himanshu.madhani@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 66c562efbf35e98c42789f73cf39ad5f28abf6be

o validate Tx queue only in case of adapters which supports
  multi Tx queue.

  This patch is to fix regression introduced in commit
  aa4a1f7df7cbb98797c9f4edfde3c726e2b3841f
  "qlcnic: Enable Tx queue changes using ethtool for 82xx Series adapter"

Signed-off-by: Himanshu Madhani <himanshu.madhani@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ethtool.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ethtool.c
index c684445..e42cebd 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ethtool.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ethtool.c
@@ -691,7 +691,7 @@ static int qlcnic_set_channels(struct net_device *dev,
    return err;
  }
 
- if (channel->tx_count) {
+ if (qlcnic_82xx_check(adapter) && channel->tx_count) {
   err = qlcnic_validate_max_tx_rings(adapter, channel->tx_count);
   if (err)
    return err;
diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index 3febdf8..dfff064 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -3645,11 +3645,6 @@ int qlcnic_validate_max_tx_rings(struct qlcnic_adapter *adapter, u32 txq)
  u8 max_hw = QLCNIC_MAX_TX_RINGS;
  u32 max_allowed;
 
- if (!qlcnic_82xx_check(adapter)) {
-  netdev_err(netdev, "No Multi TX-Q support\n");
-  return -EINVAL;
- }
-
  if (!qlcnic_use_msi_x && !qlcnic_use_msi) {
   netdev_err(netdev, "No Multi TX-Q support in INT-x mode\n");
   return -EINVAL;
@@ -3689,8 +3684,7 @@ int qlcnic_validate_max_rss(struct qlcnic_adapter *adapter,
  u8 max_hw = adapter->ahw->max_rx_ques;
  u32 max_allowed;
 
- if (qlcnic_82xx_check(adapter) && !qlcnic_use_msi_x &&
-     !qlcnic_use_msi) {
+ if (!qlcnic_use_msi_x && !qlcnic_use_msi) {
   netdev_err(netdev, "No RSS support in INT-x mode\n");
   return -EINVAL;
  }
-- 
1.7.1